### PR TITLE
Bump version to 0.4.1

### PR DESCRIPTION
**Pull request by john-odonnell**
### Desired Outcome

Release `summon-aws-secrets` at v0.4.1

### Implemented Changes

- Update version in `VERSION` and `version.go` files
- Update `bin/build` and `Jenkinsfile` to build release artifacts on tagged builds
- Update `.goreleaser.yml` to generate Homebrew formula
- Add release instructions to `CONTRIBUTING.md`

### Connected Issue/Story

CyberArk internal issue link: [ONYX-19912](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-19912)

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [x] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -14,3 +14,41 @@ For general contribution and community guidelines, please see the [community rep
...
+   and attach them to the release:
+   - `SHA256SUMS.txt`
+   - `summon-aws-secrets-x.y.z-darwin-amd64.tar.gz`
+   - `summon-aws-secrets-x.y.z-darwin-arm64.tar.gz`
+   - `summon-aws-secrets-x.y.z-freebsd-amd64.tar.gz`
+   - `summon-aws-secrets-x.y.z-linux-amd64.tar.gz`
+   - `summon-aws-secrets-x.y.z-netbsd-amd64.tar.gz`
+   - `summon-aws-secrets-x.y.z-openbsd-amd64.tar.gz`
+   - `summon-aws-secrets-x.y.z-solaris-amd64.tar.gz`
+   - `summon-aws-secrets-x.y.z-windows-amd64.tar.gz`
```
Comment (rpothier) [inline CONTRIBUTING.md:54]: Thanks for updating this section! While you're here can you also add a comment to update [homebrew](https://github.com/cyberark/homebrew-tools) tools
like [summon](https://github.com/cyberark/summon/blob/main/CONTRIBUTING.md#releasing) has?

**Review by rpothier — COMMENTED** [review]
Looks good! Addition if you don't mind.

File: CONTRIBUTING.md (hunk 1 above)
Comment (john-odonnell) [reply CONTRIBUTING.md]: Thanks for the catch. I updated the goreleaser config to generate a Formula cookbook and added to the Release instrux.

**Review by rpothier — APPROVED** [review]
LGTM!

**Pull request merged**